### PR TITLE
[PX-4220] - Track analytics for weight & framed dimensions in CMS

### PR DESCRIPTION
**Pull request by brendamanganelli**
This PR adds specific event for framed dimensions dropdown menus. 

**EDIT:**

It also includes some name changes to framed dimensions events to better align with the UI/database.

### CODE REVIEW

File: src/Schema/Events/Click.ts [hunk 1]
```diff
@@ -542,6 +542,29 @@ export interface ClickedOnFrameMeasurements {
   context_page_owner_id?: string
 }
 
+/**
+ * A Partner clicks on Artwork frame dimension dropdowns to choose either frame dimensions (H, W, D) or diameter
+ * and/or frame metric in the artwork edit page in the CMS.
+ *
+ * This schema describes events sent to Segment from [[clickedOnFrameMeasuremetsDropdown]]
```
Comment (pepopowitz) [inline src/Schema/Events/Click.ts:549]:
```suggestion
 * This schema describes events sent to Segment from [[clickedOnFrameMeasurementsDropdown]]
```

File: src/Schema/Events/Click.ts [hunk 2]
```diff
@@ -542,6 +542,29 @@ export interface ClickedOnFrameMeasurements {
...
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "clickedOnFrameMeasuremetsDropdown,
```
Comment (pepopowitz) [inline src/Schema/Events/Click.ts:554]:
```suggestion
 *    action: "clickedOnFrameMeasurementsDropdown,
```

File: src/Schema/Events/Click.ts [hunk 3]
```diff
@@ -542,6 +542,29 @@ export interface ClickedOnFrameMeasurements {
...
+ *    context_module: "voltArtworksEdit",
+ *    context_page_owner_type: "artwork",
+ *    context_page_owner_id: "60de173a47476c000fd5c4cc"
+ *  }
+ * ```
+ */
+ export interface ClickedOnFrameMeasuremetsDropdown {
```
Comment (pepopowitz) [inline src/Schema/Events/Click.ts:561]:
```suggestion
 export interface ClickedOnFrameMeasurementsDropdown {
```

File: src/Schema/Events/index.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ import {
   ClickedOnArtworkShippingUnitsDropdown,
   ClickedOnArtworkShippingWeight,
   ClickedOnFrameMeasurements,
+  ClickedOnFrameMeasuremetsDropdown,
```
Comment (pepopowitz) [inline src/Schema/Events/index.ts:35]:
```suggestion
  ClickedOnFrameMeasurementsDropdown,
```

File: src/Schema/Events/index.ts [hunk 5]
```diff
@@ -124,6 +125,7 @@ export type Event =
   | ClickedOnArtworkShippingWeight
   | ClickedOnArtworkShippingUnitsDropdown
   | ClickedOnFrameMeasurements
+  | ClickedOnFrameMeasuremetsDropdown
```
Comment (pepopowitz) [inline src/Schema/Events/index.ts:128]:
```suggestion
  | ClickedOnFrameMeasurementsDropdown
```

File: src/Schema/Events/index.ts [hunk 6]
```diff
@@ -298,6 +300,10 @@ export enum ActionType {
    * Corresponds to {@link ClickedOnFrameMeasurements}
    */
   clickedOnFrameMeasurements = "clickedOnFrameMeasurements",
+  /**
+   * Corresponds to {@link ClickedOnFrameMeasuremetsDropdown}
```
Comment (pepopowitz) [inline src/Schema/Events/index.ts:304]:
```suggestion
   * Corresponds to {@link ClickedOnFrameMeasurementsDropdown}
```

File: src/Schema/Events/index.ts [hunk 7]
```diff
@@ -298,6 +300,10 @@ export enum ActionType {
...
+   */
+   clickedOnFrameMeasuremetsDropdown = "clickedOnFrameMeasuremetsDropdown",
```
Comment (pepopowitz) [inline src/Schema/Events/index.ts:306]:
```suggestion
   clickedOnFrameMeasurementsDropdown = "clickedOnFrameMeasurementsDropdown",
```

**Review by pepopowitz — COMMENTED** [review]
Thank you! One suggested spelling correction (in multiple places 😄  )

File: src/Schema/Events/Click.ts [hunk 8]
```diff
@@ -513,30 +513,53 @@ export interface ClickedOnArtworkShippingWeight {
...
+ * A Partner clicks on framed measurements fields in the CMS artworks edit page.
  *
- * This schema describes events sent to Segment from [[clickedOnFrameMeasurements]]
+ * This schema describes events sent to Segment from [[clickedOnFramedMeasurements]]
  *
  *  @example
  *  ```
  *  {
- *    action: "clickedOnFrameMeasurements",
+ *    action: "clickedOnFramedMeasurements",
```
Comment (pepopowitz) [inline src/Schema/Events/Click.ts:531]: @brendamanganelli this is the only change we made that we wanted to call out to you for confirmation -- we changed all terminology from "frame measurements" to "framed measurements" (note the `d` in framed) to align with how the UI/database have been referring to them. Let us know if this change is problematic for any of your work.

**Review by pepopowitz — APPROVED** [review]
@brendamanganelli - @annacarey and I noticed a few small details we wanted to change/fix, so we included it in this PR. We're merging now to move forward, but let us know if any of our changes need to be reverted!

**Comment by artsyit** [comment]
:rocket: PR was released in [`v2.15.0`](https://github.com/artsy/cohesion/releases/tag/v2.15.0) :rocket:

**Pull request merged**